src/plugins/workspace/public/components/global_search/search_pages_command.tsx

```diff
@@ -36,50 +51,61 @@ export const searchPages = async (
-        return links
-          .filter((link) => {
-            const title = link.title;
-            let parentNavLinkTitle;
-            // parent title also taken into consideration for search its sub items
-            if (link.parentNavLinkId) {
-              parentNavLinkTitle = navGroup.navLinks.find(
-                (navLink) => navLink.id === link.parentNavLinkId
-              )?.title;
-            }
-            const titleMatch = title && title.toLowerCase().includes(query.toLowerCase());
-            const parentTitleMatch =
-              parentNavLinkTitle && parentNavLinkTitle.toLowerCase().includes(query.toLowerCase());
-            return (
-              !link.hidden &&
-              !link.disabled &&
-              (titleMatch || parentTitleMatch) &&
-              !parentNavLinkIds.includes(link.id)
-            );
-          })
-          .map((link) => ({
-            ...link,
-            navGroup,
-          }));
+    const searchResult = searchNavigationLinks(allAvailableCaseId, navGroupMap, query);
+
+    const handleCallback = (link: Link) => {
+      callback?.();
+      const isPageOutOfWorkspace = link.navGroup.type === NavGroupType.SYSTEM;
+      if (isPageOutOfWorkspace && currentWorkspace) {
+        // remove workspace information in the URL, special handling for data source which could visible both in/out workspace
+        const urlWithoutWorkspace = formatUrlWithWorkspaceId(
+          link.href,
+          '',
+          coreStart.http.basePath
+        );
+        window.location.assign(urlWithoutWorkspace);
+        return;
       }
-      return [];
-    });
+      coreStart.application.navigateToApp(link.id);
+    };
+
+    const renderBreadcrumbs = (
+      link: Link,
+      breadcrumbs: EuiBreadcrumb[],
+      availableUseCases: WorkspaceUseCase[]
+    ) => {
+      const isPageOutOfWorkspace = link.navGroup.type === NavGroupType.SYSTEM;
+
+      const updatedBreadcrumbs = [...breadcrumbs];
+      if (currentWorkspace && !isPageOutOfWorkspace) {
+        updatedBreadcrumbs.push({
+          text: (
+            <WorkspaceTitleDisplay
+              workspace={currentWorkspace}
+              availableUseCases={availableUseCases || []}
+            />
+          ),
+        });
+      } else {
+        updatedBreadcrumbs.push({ text: navGroupElement(link.navGroup) });
+      }
+
+      return updatedBreadcrumbs;
+    };
 
     const pages = searchResult.slice(0, 10).map((link) => {
       return (
         <GlobalSearchPageItem
           link={link}
           search={query}
-          currentWorkspace={currentWorkspace}
           application={coreStart.application}
-          http={coreStart.http}
-          registeredUseCases$={registeredUseCases$}
-          callback={callback}
+          callback={() => {
+            handleCallback(link);
+          }}
+          renderBreadcrumbs={(breadcrumbs) => {
+            const availableUseCases = useObservable(registeredUseCases$);
```

Comment:
I'm not prefer to execute a react hook in side render function. Although it work as expected for now. But it's not a good practice. Maybe we need to create a `WorkspaceGlobalSearchPageItem` component.

Reply:
Done, created a new `WorkspaceGlobalSearchPageItem `